test/dataframes.jl

```diff
@@ -78,6 +78,9 @@ const ≅ = isequal
     @test @transform(df, n = 1).n == fill(1, nrow(df))
 
     @test @transform(df, n = :i .* :g).n == [1, 2, 3, 8, 10]
+    
+    @transform!(df, n = :i)
```

Comment:
For tests can you please do the following? 

1. Copy and paste all the `@transform` tests but add `@transform!`
2. Add some tests in addition to those above which test for `===` on columns in the DataFrame.